src/pages/index.js

```diff
@@ -1,101 +1,245 @@
-const editFormValidator = new FormValidator(config, profileForm);
-addFormValidator.enableValidation();
-editFormValidator.enableValidation();
+const avatarForm = document.forms["edit-avatar-form"];
+
+/* -------------------------------------------------------------------------- */
+/*                                  Api                                   */
+/* -------------------------------------------------------------------------- */
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "eac0c1a2-60c5-44c4-979d-14695d75b6b5",
+    "Content-Type": "application/json",
+  },
+});
 
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
 section.renderItems();
 
-const popupEditForm = new PopupWithForm(
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.setItems(cards);
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+/* -------------------------------------------------------------------------- */
+/*                                  Popups                                    */
+/* -------------------------------------------------------------------------- */
+const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
-popupEditForm.setEventListeners();
+editProfilePopup.setEventListeners();
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
 
-const popupAddForm = new PopupWithForm(
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   handleAddCardSubmit
 );
-popupAddForm.setEventListeners();
+addCardPopup.setEventListeners();
 
 const popupImage = new PopupWithImage("#modal__image");
-popupImage.setEventListiners();
+popupImage.setEventListeners();
 
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
 // FUNCTIONS/////
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getCardElement();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleLikeClicks,
+    handleDeleteSubmit
+  );
+  return card.getView();
 }
 
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+function handleImageClick(cardData) {
+  popupImage.open(cardData);
 }
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  popupEditForm.close();
+function handleAvatarSubmit(url) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(res);
+      avatarForm.reset();
+      avatarFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs. But when you change the avatar, you don't have any errors after submitting. You need to call `disableButton` there.